packages/core-transaction-pool-mem/lib/connection.js

```diff
@@ -0,0 +1,340 @@
+   * @param  {Array} transactions
+   * @return {void}
+   */
+  removeTransactions (transactions) {
+    transactions.forEach(t => this.removeTransaction(t))
+  }
+
+  /**
+   * Check whether sender of transaction has exceeded max transactions in queue.
+   * @param  {Transaction} transaction
+   * @return {Boolean} true if exceeded
+   */
+  hasExceededMaxTransactions (transaction) {
+    this.__purgeExpired()
+
+    if (this.options.allowedSenders.includes(transaction.senderPublicKey)) {
+      logger.debug(
+        `Transaction pool: allowing sender public key: ${transaction.senderPublicKey} ` +
+        '(listed in options.allowedSenders), thus skipping throttling.')
+      return false
+    }
+
+    const count = this.mem.getIdsBySender(transaction.senderPublicKey).size
+
+    return !(count <= this.options.maxTransactionsPerSender)
+  }
+
+  /**
+   * Get a transaction by transaction id.
+   * @param  {String} id
+   * @return {(Transaction|undefined)}
+   */
+  getTransaction (id) {
+    this.__purgeExpired()
+
+    return this.mem.getTransactionById(id)
+  }
+
+  /**
+   * Removes any transactions in the pool that have already been forged.
+   * @param  {Array} transactionIds
+   * @return {Array} IDs of pending transactions that have yet to be forged.
+   */
+  async removeForgedAndGetPending (transactionIds) {
+    const forgedIdsSet = new Set(await database.getForgedTransactionsIds(transactionIds))
+
+    forgedIdsSet.forEach(id => this.removeTransactionById(id))
+
+    return transactionIds.filter(id => !forgedIdsSet.has(id))
+  }
+
+  /**
+   * Get all transactions that are ready to be forged.
+   * @param  {Number} blockSize
+   * @return {(Array|void)}
+   */
+  async getTransactionsForForging (blockSize) {
+    this.__purgeExpired()
+
+    try {
+      let transactionsIds = await this.getTransactionIdsForForging(0, this.mem.getSize())
+
+      let transactions = []
+      while (transactionsIds.length) {
+        const id = transactionsIds.shift()
+        const transaction = this.mem.getTransactionById(id)
+
+        if (!transaction ||
+            !this.checkDynamicFeeMatch(transaction) ||
+            !(await this.checkApplyToBlockchain(transaction))) {
+          continue
+        }
+
+        transactions.push(transaction.serialized)
+        if (transactions.length === blockSize) {
+          break
+        }
+      }
+
+      return transactions
+    } catch (error) {
+      logger.error('Could not get transactions for forging: ', error, error.stack)
+    }
+  }
+
+  /**
+   * Get all transactions within the specified range (in insertion order).
+   * @param  {Number} start
+   * @param  {Number} size
+   * @return {(Array|void)} array of serialized transaction hex strings
+   */
+  getTransactions (start, size) {
+    this.__purgeExpired()
+
+    let serializedTransactions = []
+
+    let i = 0
+    for (let transaction of this.mem.getTransactionsInInsertionOrder()) {
```

Comment:
Minor nitpick, but this `for loop` is the basically same as in `getTransactionIdsForForging` could be moved into it's own function. It would just need to know which `property` to pick.